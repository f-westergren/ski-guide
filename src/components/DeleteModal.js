import React, { useState } from 'react';
import { Button, Modal, ModalFooter } from 'reactstrap';

const DeleteModal = ({ btnText, remove, className='' }) => {
  const [modal, setModal] = useState(false);

  const toggle = () => {
    setModal(!modal);
  }

  const handleClick = () => {
    remove();
    toggle();
  }

  return(
    <>
    <Button className={className} color="danger" onClick={toggle}>{btnText}</Button>
    <Modal isOpen={modal} toggle={toggle}>
      <h4 className="text-center">Are you sure?</h4>
      <ModalFooter>
        <Button color="danger" onClick={handleClick}>Delete</Button>
        <Button outline color="danger" onClick={toggle}>Cancel</Button>
      </ModalFooter>
    </Modal>
    </>
  )
}

export default DeleteModal;