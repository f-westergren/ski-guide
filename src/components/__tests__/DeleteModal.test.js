import React from "react";
import { render } from "@testing-library/react";
import DeleteModal from "../DeleteModal";

it("renders without crashing", function() {
  render(<DeleteModal />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<DeleteModal />);
  expect(asFragment()).toMatchSnapshot();
});
