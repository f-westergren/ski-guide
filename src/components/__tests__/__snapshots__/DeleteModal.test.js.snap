// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-danger"
    type="button"
  />
</DocumentFragment>
`;
